cephfs/file.go

```diff
@@ -199,3 +210,30 @@ func (f *File) Fchown(user uint32, group uint32) error {
 	ret := C.ceph_fchown(f.mount.mount, f.fd, C.int(user), C.int(group))
 	return getError(ret)
 }
+
+// Fsync ensures the file content that may be cached is committed to stable
+// storage.
+// Pass SyncMetaData to have this call behave like standard fsync.
+// Pass SyncDataOnly to have this call behave more like fdatasync (linux).
+//
+// Implements:
+//  int ceph_fsync(struct ceph_mount_info *cmount, int fd, int syncdataonly);
+func (f *File) Fsync(sync SyncChoice) error {
+	if err := f.validate(); err != nil {
+		return err
+	}
+
+	ret := C.ceph_fsync(
+		f.mount.mount,
+		f.fd,
+		C.int(sync),
+	)
+	return getError(ret)
+}
+
+// Sync ensures the file content that may be cached is committed to stable
+// storage.
+// Sync behaves like Go's os package File.Sync function.
+func (f *File) Sync() error {
```

Comment:
If you add this, there should also be a plan to add `Chmod()` and similar that make a `cephfs.File` behave more like an `os.File`... Just wondering what the plan is here.

Reply:
Er, I was afraid you'd point something like this out. I'm totally  ok with eventually adding more "compatibility" methods to the File type. In this case I thought that ceph's Fsync was a bit clear about what to do and felt a simple wrapper to nudge users of the library in the right direction unless they have a more specialized use case (I read a man page so you don't have too!)

I'll also point out that go interfaces don't have to be maximal so you could have something more akin to [io.WriteCloser](https://golang.org/pkg/io/#WriteCloser), an io.WriteSyncerCloser [1] perhaps, without having to worry about Chmod/Chown/truncate/whatever-else.

1 - Yes I have a use case for such a thing. If you're curious just ask! :-)

Reply:
> I ... felt a simple wrapper to nudge users of the library in the right direction unless they have a more specialized use case

That is something I support most definitely. But I prefer to get that done in a more planned way, maybe in its own `go-ceph/cephfs/simple` package or something.